client/me/purchases/remove-plan-dialog/index.tsx

```diff
@@ -0,0 +1,213 @@
+import { Dialog, Gridicon } from '@automattic/components';
+import { createElement, createInterpolateElement } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import { Fragment } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { SiteScreenshot } from '../site-screenshot/';
+import getPlanFeatures from './get-plan-features';
+import './style.scss';
+
+interface RemovePlanDialogProps {
+	closeDialog: () => void;
+	removePlan: () => void;
+	isDialogVisible: boolean;
+	isRemoving: boolean;
+	site: SiteExcerptData;
+	hasDomain: boolean;
+	isRefundable: boolean;
+	isAutoRenewing: boolean;
+	primaryDomain: string;
+	wpcomURL: string;
+}
+
+export const RemovePlanDialog = ( {
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	site,
+	hasDomain,
+	isRefundable,
+	isAutoRenewing,
+	primaryDomain,
+	wpcomURL,
+}: RemovePlanDialogProps ) => {
+	const translate = useTranslate();
+	const siteName = site.name ?? '';
+
+	/**
+	 * Istantiate site's plan variables.
+	 */
+	const productSlug = site.plan?.product_slug;
+	const planLabel = site.plan?.product_name_short;
+	const isComingSoon = site.is_coming_soon;
+	const isPrivate = site.is_private;
+	const launchedStatus = site.launch_status === 'launched' ? true : false;
+	const shouldUseSiteThumbnail =
+		isComingSoon === false && isPrivate === false && launchedStatus === true;
+	let subTitle = '';
+
+	if ( isRefundable && isAutoRenewing ) {
+		subTitle = translate( 'If you cancel your plan, you will lose:' );
+	} else if ( ! isRefundable && isAutoRenewing ) {
+		subTitle = translate( 'If you cancel your plan, once it expires, you will lose:' );
+	} else if ( ! isRefundable && ! isAutoRenewing ) {
+		translate( 'If you remove your plan, you will lose:' );
```

Comment:
> Not sure if I'm doing something wrong, but I'm still seeing problems for a non-refundable subscription with auto-renew on:
> 
> * Going through the cancellation flow still results in the plan being removed rather than auto-renew being turned off.
> * I also noticed the "Back" button inside the cancellation survey was broken in this case (clicking it did nothing).
> 
> In addition, the intro paragraph never seems to be displaying in the modal currently (see inline comment for more about that).
> 

I fixed a bug and the 3 bugs above look resolved to me.

> Besides the above, it looks like it's mostly working, but the jumping in and out of different modals while cancelling still seems disjointed to me. @Automattic/shilling should perhaps take a closer look at the overall flow next week. I'd post screenshots or a video but don't have time right now.

I'm progressing on the work for this case: https://dotcomretention.wordpress.com/2022/09/12/cancelation-modal-edge-cases/ it will be covered by a different PR and this should be the only disjointed case. For all the other cases the Cancellation Modal shows the survey form without disjunctions.